crates/webidl/src/util.rs

```diff
@@ -294,9 +296,24 @@ impl<'src> FirstPassRecord<'src> {
                 }
             },
         };
-        let js_ret = ret.clone();
+
+        let is_indexing_getter_to_fix = is_indexing_getter && // exclude non index getters.
+            !catch && // exclude indexing getters that return `Result<T>`.
+            if let Some(ref ty) = ret { // exclude indexing getters that return `Option<T>`.
+                !format!("{}", quote! { #ty })
+                    .replace(" ", "")
+                    .starts_with("Option<")
```

Comment:
Help: I think there should be a smarter way to tell whether `ty` is a `Option< ... >` or not.